### PR TITLE
[Stroage] Delete stale state values and merkle nodes at version 0 after fast sync.

### PR DESCRIPTION
**Pull request by grao1991**

### CODE REVIEW

**Review by msmouse — APPROVED** [review]
Deleting the genesis is really messy from the first place. We should really try to fast sync to an empty DB instead.. 
cc @JoshLind

**Comment by msmouse** [comment]
Okay, guess we need to talk with @JoshLind  to see if we can make the right way -- start fast syncing to an empty DB in the first place

File: storage/aptosdb/src/lib.rs [hunk 1]
```diff
@@ -2070,6 +2068,37 @@ impl DbWriter for AptosDB {
...
+    let mut iter = state_merkle_db.iter::<JellyfishMerkleNodeSchema>(ReadOptions::default())?;
+    iter.seek(&NodeKey::new_empty_path(0))?;
+    for item in iter {
+        let (node_key, node) = item?;
+
+        if node_key.version() != 0 {
+            break;
+        }
+
+        let node_key_at_fast_sync_version =
```
[the remark this answers is not in the document]
Comment (grao1991) [reply storage/aptosdb/src/lib.rs]: The previous logic has some issues.. updated now.

File: storage/aptosdb/src/lib.rs [hunk 2]
```diff
@@ -2070,6 +2069,47 @@ impl DbWriter for AptosDB {
...
+
+        if let Node::Leaf(leaf_node) = node {
+            let state_key = leaf_node.value_index().0.clone();
+            let (version, _) = StateDb::get_state_value_with_version_by_version_ext(
+                state_kv_db,
+                &state_key,
+                fast_sync_version,
+            )?
+            .expect("Must exist.");
+            if version != 0
```
Comment (areshand) [inline storage/aptosdb/src/lib.rs:2097]: would be great to add some documentation on the reasoning of having this condition?
My understanding is:
version != 0 seems to indicate the node value has been updated after fast_sync version
version == 0 means the value hasn't been updated. 
why do we only want to delete the nodes in one scenario?

**Review by davidiw — APPROVED** [review]

**Review by davidiw — CHANGES_REQUESTED** [review]
we can't leave approved PRs open this long...

**Pull request closed without merging**